packages/agreement-email-sender/src/services/agreementEmailSenderService.ts

```diff
@@ -0,0 +1,170 @@
+  );
+
+  const [eservice, producer, consumer] = await Promise.all([
+    getEServiceById(agreement.eserviceId),
+    getTenantById(agreement.producerId),
+    getTenantById(agreement.consumerId),
+  ]);
+
+  if (!eservice) {
+    throw eServiceNotFound(agreement.eserviceId);
+  }
+
+  if (!producer) {
+    throw genericInternalError(
+      `Produce tenant not found for agreement ${agreement.id}`
+    );
+  }
+
+  if (!consumer) {
+    throw genericInternalError(
+      `Consumer tenant not found for agreement ${agreement.id}`
+    );
+  }
+
+  const producerSelfcareId = producer.selfcareId;
+  if (!producerSelfcareId) {
+    throw selfcareIdNotFound(producer.id);
+  }
+
+  const consumerSelfcareId = consumer.selfcareId;
+  if (!consumerSelfcareId) {
+    throw selfcareIdNotFound(consumer.id);
+  }
+
+  const producerInstitution = await getInstitution(
+    producerSelfcareId,
+    selfcareV2Client
+  );
+
+  const consumerInstitution = await getInstitution(
+    consumerSelfcareId,
+    selfcareV2Client
+  );
+
+  const producerEmail = producerInstitution?.digitalAddress;
+  const consumerEmail = consumerInstitution?.digitalAddress;
+
+  if (!producerEmail) {
+    throw genericInternalError(
+      `Producer digital address not found for agreement ${agreement.id}`
+    );
+  }
+
+  if (!consumerEmail) {
+    throw genericInternalError(
+      `Consumer digital address not found for agreement ${agreement.id}`
+    );
+  }
+
+  const descriptor = eservice.descriptors.find(
+    (d) => d.id === agreement.descriptorId
+  );
+
+  if (!descriptor) {
+    throw descriptorNotFound(agreement.eserviceId, agreement.descriptorId);
+  }
+
+  return {
+    subject: `Richiesta di fruizione ${agreement.id} attiva`,
+    to: [producerEmail, consumerEmail],
+    body: templateService.compileHtml(htmlTemplate, {
+      activationDate: formattedActivationDate,
+      agreementId: agreement.id,
+      eserviceName: eservice.name,
+      eserviceVersion: descriptor.version,
+      producerName: producer.name,
+      consumerName: consumer.name,
+    }),
+  };
+}
+
+export async function sendAgreementEmail(
+  agreement: AgreementV2,
+  readModelService: ReadModelService,
+  selfcareV2Client: SelfcareV2Client,
+  emailManager: EmailManager,
+  { agreementEmailSender } = agreementEmailSenderConfig()
+): Promise<void> {
+  const { to, subject, body } = await getActivationMailFromAgreement(
+    agreement,
+    readModelService,
+    selfcareV2Client
+  );
+
+  await emailManager.send(agreementEmailSender, to, subject, body);
+}
+
+async function getInstitution(
+  id: string,
+  selfcareV2Client: SelfcareV2Client
+): Promise<InstitutionResponse> {
+  try {
+    return await selfcareV2Client.getInstitution({
+      params: { id },
+    });
+  } catch (error) {
+    const code = mapInstitutionError(error);
+    if (code === 404) {
+      throw institutionNotFound(id);
+    }
+    throw genericInternalError(`Error getting institution ${id}`);
```

Comment:
very very minor consideration: we could add the selfcare error in our internal message and avoid passing the logger

Reply:
I am already working on the improvement to remove the selfcare calls as you suggested, so this will soon go away anyways :) https://github.com/pagopa/interop-be-monorepo/pull/633